lib/gst/player/gstplayer.c

```diff
@@ -2617,13 +2623,24 @@ gst_player_seek_internal_locked (GstPlayer * self)
   remove_tick_source (self);
   self->is_eos = FALSE;
 
-  ret =
-      gst_element_seek_simple (self->playbin, GST_FORMAT_TIME,
-      GST_SEEK_FLAG_FLUSH, position);
+  flags |= GST_SEEK_FLAG_FLUSH;
+  flags |= GST_SEEK_FLAG_ACCURATE;
 
-  if (!ret)
-    emit_error (self, g_error_new (GST_PLAYER_ERROR, GST_PLAYER_ERROR_FAILED,
+  if (self->trick_play)
+    flags |= GST_SEEK_FLAG_TRICKMODE;
```

Comment:
Maybe we want to expose this as a property? Selecting what kind of trick mode we want: a) full and accurate, b) allow dropping of non-keyframes, c) drop keyframes, d) disable audio (which actually makes sense together with each of a) b) and c))